common/cache/lru.go

```diff
@@ -330,6 +332,10 @@ func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool)
 				c.currSize = newCacheSize
 				metrics.CacheUsage.With(c.metricsHandler).Record(float64(c.currSize))
 				c.updateEntryTTL(existingEntry)
+
+				if callback, ok := value.(PutCallback); ok {
+					callback.OnPut()
+				}
```

Comment:
This code might need a refactor that's out of scope of this PR. Therefore, this code needed to be added twice.